### PR TITLE
Frost and MuSig joint key Point type parameter

### PR DESCRIPTION
**Pull request by nickfarrow**
* Change musig and frost keys to allow for any point type
* keygen creates Normal points
* signing operations are done with EvenY points

### CODE REVIEW

File: schnorr_fun/src/frost.rs [hunk 1]
```diff
@@ -473,30 +438,50 @@ impl<H: Digest<OutputSize = U32> + Clone, NG: NonceGen> Frost<H, NG> {
...
-            session_id,
-            public_key,
-            message,
-        )
+    ) -> R
+    where
+        <R as SeedableRng>::Seed: From<GenericArray<u8, U32>>,
+    {
+        let message = message.unwrap_or(Message::raw(b""));
+        let msg_len = (message.len() as u64).to_be_bytes();
```
Comment (nickfarrow) [inline schnorr_fun/src/frost.rs:467]: These could be moved into the macro?

File: schnorr_fun/tests/frost_prop.rs [hunk 2]
```diff
@@ -54,13 +56,18 @@ proptest! {
         let sid = b"frost-prop-test".as_slice();
         let message = Message::plain("test", b"test");
 
-        let nonces: Vec<_> = signer_indexes.iter().map(|i|
-            proto.gen_nonce(
-                &secret_shares[*i],
+        let mut nonce_rngs: Vec<ChaCha20Rng> = secret_shares.iter().map(|secret_share| {
```
Comment (nickfarrow) [inline schnorr_fun/tests/frost_prop.rs:59]: `RngAlgorithm::ChaCha` is not `Seedable` so using `ChaCha20Rng`... Doesn't seem ideal

**Comment by LLFourn** [comment]
Sorry I somehow goofed the thing when merged the parent PR and deleted the branch. Can you reopen?

File: schnorr_fun/tests/frost_prop.rs (hunk 2 above)
Comment (LLFourn) [reply schnorr_fun/tests/frost_prop.rs:59]: `ChaCha20Rng` is the right choice.

File: schnorr_fun/src/frost.rs (hunk 1 above)
Comment (LLFourn) [reply schnorr_fun/src/frost.rs:467]: it could be moved into `HashAdd`.

**Pull request closed without merging**